config/config.go

```diff
@@ -30,13 +31,37 @@ const (
 
 	EnvFnRegistry = "registry"
 	EnvFnContext  = "context"
+
+	OCI_CLI_AUTH_ENV_VAR            = "OCI_CLI_AUTH"
+	OCI_CLI_AUTH_INSTANCE_PRINCIPAL = "instance_principal"
+	OCI_CLI_AUTH_INSTANCE_OBO_USER  = "instance_obo_user"
 )
 
-var defaultRootConfigContents = &ContextMap{CurrentContext: "", CurrentCliVersion: ""}
-var DefaultContextConfigContents = &ContextMap{
-	ContextProvider:      DefaultProvider,
-	provider.CfgFnAPIURL: defaultLocalAPIURL,
-	EnvFnRegistry:        "",
+var defaultRootConfigContents = &ContextMap{CurrentContext: "default", CurrentCliVersion: Version}
+
+func DefaultContextConfigContents() (contextMap *ContextMap) {
+	ociCliAuth := os.Getenv(OCI_CLI_AUTH_ENV_VAR)
+
+	if ociCliAuth == OCI_CLI_AUTH_INSTANCE_OBO_USER {
```

Comment:
nit: could use `switch (ociCliAuth)` (makes it clear it's case analysis, not arbitrary conditional logic)